### DISCUSSION TITLE
width widgets for wordpress elementor

### DISCUSSION THREAD

**Discussion by dacunha31** [elementor/elementor#18001] [Developers Community]
Hi,
Elementor Team sent me here to ask for help.
I am new to website creation and I am currently creating a website for my son who is going to open his business in France. This website is currently hosted locally by WampServer and I use WordPress, Elementor and the AERA theme to design it. I have a side widget area that I would like to reduce in order to leave more space for the body of the site. I don’t know how to do it. I suppose that a CSS could help me but as I explained to you, I am a novice. Could you help me? I will add screenshot to help understand. I would also like the widget side area (where the buttons are) to be able to drop down as the person scrolls. 
I can of course use Elementor as an editor but I don't have a sidebar to host my widgets... Unless a css code allows it? 
Thanks for your help.
https://www.cjoint.com/c/LBvlhKwLjvh

**Comment by m1ga** [comment]
The screenshot doesn't really help as the HMTL structure is the interesting part. Looking at the default AERA theme you can override it with something like:
```css
#post-section #av-primary-content {
	width: 90%;
	max-width: 90%;
}
#post-section #av-secondary-content {
	width: 10% ;
	max-width: 10%;
}
```
(just a quick hack by looking at the structure for 10sec).

But this discussion board is more about Elementor and not Wordpress/HTML development in general. It would be best to ask e.g. in the AERA support page.

**Reply by dacunha31** [reply]
Yes I understand that it is not easy.
What you suggest, I have to put it in the additional css?
Where can I find the AERA support page? 
Aera told me that my problem was with Elementor and then they said that if I paid they could look into it further...
So I don't know if it's Elementor or AERA from Nayra.

**Reply by m1ga** [reply]
Perhaps you should start with some basic Wordpress tutorials :wink: You can add additional CSS in `Themes - Customize - Additional CSS`: https://www.wpbeginner.com/plugins/how-to-easily-add-custom-css-to-your-wordpress-site/
And since AERA defines the sidebar and content in your case its the theme. And a simple google search will bring you to https://www.nayrathemes.com/aera-free/ - there is a "need help" button. 
There are so many ways to create the layout so it can be a theme issue, a wordpress issue or an Elementor issue. Just depends on how you start building the page.

**Reply by dacunha31** [reply]
Yes I'm a novice but I can manage as soon as someone explains it to me.
For now, I removed all the widgets in the side column and so my page becomes wider again. I have created in each page 2 columns and I put in my right column buttons (identical to those I had created with my widgets) and I keep a correct page width. The drawback of this solution is that I have to create in each page this column with the 3 buttons... it's not ideal.
I've already added additional css in the area you're talking about for other changes.
I have contacted Nayra several times and they don't give me any solution for this problem.
I created this site with the AERA theme but also with Elementor.

**Reply by m1ga** [reply]
If you buy the [Elementor Pro version](https://trk.elementor.com/26724) you can define page templates and reuse them all the time. You could also save the page as a custom template and reuse that when creating a page. Check some Elementor tutorials on templating.

**Reply by m1ga** [reply]
Also: if you use a different theme e.g. Astra you can just resize the sidebar with a slider in the theme customizer. Perhaps Aera might not be the best starting theme for you

**Reply by dacunha31** [reply]
I created this site for my young son (20 years old) who is starting his career as an entrepreneur. His current finances do not allow him to subscribe to paid extensions for the moment. That's why we are looking for other solutions.
With the css codes, I managed to change a lot of things, except the size of the widget area.

**Reply by dacunha31** [reply]
If I change my theme now, will everything on the site change?

**Reply by m1ga** [reply]
>  With the css codes, I managed to change a lot of things, except the size of the widget area.

The CSS code to do that is about :wink: Use that instead of defining the thinks in every page all over again. There a many ways to do even with the free version

**Reply by m1ga** [reply]
> If I change my theme now, will everything on the site change?

Just try it, you've said it's a local page so it's only visibile to you. And since its a theme switch it shouldn't affect your content. But I'll stop here now. This has nothing todo with Elementor and shouldn't be here in the first place :smile: Good luck, think you're on the right way to solve it

**Reply by dacunha31** [reply]
I tried to put this code in the additional css area but it doesn't work.

#post-section #av-primary-content {
	width: 90%;
	max-width: 90%;
}
#post-section #av-secondary-content {
	width: 10% ;
	max-width: 10%;
}
So I will not be able to solve this problem. Thanks for your help.
However, I have another problem that Nayra (AERA) told me came from Elementor, it is the effects (transitions) that occur at each page opening. I would like to remove them because it gives the impression of loading the same page twice and it's really not pleasant.

**Reply by m1ga** [reply]
![Screenshot_20220227_190539](https://user-images.githubusercontent.com/4334997/155894230-9244b594-cf61-45a0-a055-c9a360132667.png)
working fine here. As said earlier: your screenshot doesn't help, the structure would be better to see the classes. These are the default classes for me when I use the theme and add a widget to the sidebar. They are 2/3 - 1/3 before and with the CSS it will be 90% - 10%.

There are no default transitions in Elementor unless you enable them. So the "slide in" parts that you perhaps see come from the theme. Unless you've added them yourself in Elementor. 
I've never used that theme so I'm not a good help here either :smile:

**Reply by dacunha31** [reply]
I don't know why it works for you and not for me... I see that when I put the css in additional css, the central part of my site decreases a little (90%) but when I put a widget, this last one comes to crush the central zone which becomes very small.
![image](https://user-images.githubusercontent.com/100515539/155894518-e667376f-d525-495c-b83c-cb39942150dd.png)
For the animations, I did not add any animations on Elementor, I checked and I did not see anything.
Also, Nayra told me that it came from Elementor.
They made me remove a line in: Plugins -> Plugin Editor ->Clever Fox -> inc -> april --> april.php -> line no. 29

Remove folloiwng line

wp_enqueue_style('animate',CLEVERFOX_PLUGIN_URL .'/inc/assets/css/animate.css');
This allowed me to remove the animation on the home page only but not on the other pages.
Moreover, with the updates of the theme, the line is recreated and the animation is again present on the home page.
Here is what Nayra answered me:
This animation coming from Elementor plug-in, we already off the theme animation

**Reply by m1ga** [reply]
https://wordpress.org/plugins/clever-fox/ is a plugin by Nayra, so why should it be an Elementor issue :man_shrugging: 

You don't use the normal widget column in the theme, you use it in Elementor (you only have one `av-column-12`). 

But thats really the last post from me now :laughing:

**Reply by dacunha31** [reply]
I don't know how to tell you... that's how Nayra justifies me...
It's a pity that you can't help me more, I would have been so happy to be able to finally reach my goal.
I thank you anyway.

**Reply by dacunha31** [reply]
![image](https://user-images.githubusercontent.com/100515539/155898713-260045d8-8c71-4ff7-8ccd-8441ec882876.png)
Maybe for the animation it is Elementor?

**Comment by dacunha31** [comment]
Hello,
I tried again the code you gave me and it works partially.
Last night, I had tried it only by consulting a page where there is a form.
This morning I tried it again on the home page and it works perfectly.
I wanted to tell you that and to thank you.
The only pages where it doesn't work are the pages where there are forms (Contact From 7), pages where there are 3 columns in the main part as well as the pages where I put my pictures.
For these 3 types of pages, the widget area does not shift.
If you would be so kind as to finalize this problem with me, I would be grateful.
I filmed it to show you because it's not easy to explain in writing.
Also, on this video, you can see the animation that gives the impression of reloading the page twice. You can also see that the widgets are put in their place only at the end of the loading.
I questioned Nayra again and she told me that this is a problem with Elementor...
I don't know what to do, I really need your help.
If I could help someone in the future, I will do it with great pleasure, whether it is a question directly related to the forum or not.
https://www.cjoint.com/c/LBCjUciKsQh
Thank you for your help
![image](https://user-images.githubusercontent.com/100515539/155968438-0456eaf8-d52d-45f8-8174-c01b51512518.png)

![image](https://user-images.githubusercontent.com/100515539/155961350-94944a06-c7bb-46a7-ab5a-d18c5c2a5809.png)

**Reply by dacunha31** [reply]
I found my solution by rapoort to widget (with forms ...). I copy a page that works well and I redo my form page... and there, everything is ok !
I just have to find the solution for the Elementor animations/transitions.

**Reply by m1ga** [reply]
The theme writes `<div class="av-columns-area wow fadeInUp">` fix on your pages. Again: the theme, not Elementor! You'll find it in the Avril (parent theme of Aera) source code. 
The class name is used by Elementor since they've didn't prefix it. So you could override it with:
```
.fadeInUp {
  animation-name: none !important;
}
```

Also a little design advice: use less colors, a different font, no background behind headline. Look at other sites in that business area and incorporate that. If you son wants to use that to gain customers it has to look a bit more professional.

**Reply by dacunha31** [reply]
Thank you for your answer and especially thank you for agreeing to help me because Nayra does not want to know anything and confirms each time that it is a problem with Elementor.
I have applied the css code. It's much better but the image still jumps and the window doesn't open right away in its proper width.
https://www.cjoint.com/c/LBCnDo8hZjh
Thanks also for your advice on the design of the site.
In France, I didn't find any site dealing with the same subject. If you have a link to give me, I would be delighted. It will allow me to see the colors used.
For the colors, I think you are talking about the color of the buttons, is that right? I was thinking of putting the first one in red and the other two in blue (the same dark blue as on the menu bar).
For the background image, it reminds the activity and it seemed important to us to keep it because from the arrival on the site, the visitor can realize what this type of drone looks like!
For the font, do you mean the one used on the logo or the one used for the texts? The one for the texts is original AERA, I did not modify it. What is the recommended font and how to change it?
Thank you very much for your help.

**Reply by m1ga** [reply]
just a quick google for `drone service occitanie` brought up 
https://www.ariegedroneservices.fr/
https://www.droneservicesetagriculture.com/
https://drone-aveyron-services.fr/

Your header image has a white border around it (fixed image width?) and all elements are in an image I guess. Not good for responsive webdesign. It looks like a page from the 90s done with Frontpage :smile: 
The whole composition looks cheap/unprofessional since there is no real thought behind the structure. Check the demo page of your theme: https://www.nayrathemes.com/demo/lite/aera/
maybe use that as a starting point. Your client are people who are interested in drones or impressive images. So make a modern design with larger images or even videos (check the pages above).

**Reply by dacunha31** [reply]
Thanks for the links and comments.
The white band around the image (logo) is not mine... it is part of the Aera theme. I added a css code to enlarge the log because otherwise it's worse.
I am not able to create a site like the ones you show me... I guess they used a lot of css code to create them... I saw that one was created with wix... maybe it's better and more professional than WP and Elementor?

All elements are in an image I guess ==> What elements are you talking about? Except for the header image that I made outside the site (which is therefore an image), everything else is created on the site directly... I don't understand your sentence.

It looks like a page from the 90s done with Frontpage ==> What looks like the 90's?

Check the demo page of your theme : ==> I don't know why I don't have the same thing as on the demo page because I took the theme and I started my page creation... Unless Elementor comes to change the appearance?

maybe use that as a starting point : ==> I'll try but I don't know if I can create a second site on WP so as not to overwrite my current site

Thank you for these reviews. I will try to do better but I am not a webdesigner and I do with my financial means.

If I redo everything, should I also consider changing the theme?

Could you answer my first question about the look of the page?
I have applied the css code. It's much better but the image still jumps and the window doesn't open right away in its proper width.
https://www.cjoint.com/c/LBCnDo8hZjh

Thanks

**Reply by m1ga** [reply]
> All elements are in an image I guess ==> What elements are you talking about?

The header image. Your logo with the text above/below it and the text next to it. 

> What looks like the 90's

the whole composition. E.g. the white-blue gradients in the background of the headlines, the fade out colors in the red, yellow, green buttons on the right side

> Unless Elementor comes to change the appearance

Mentioned that before: look at tutorials. Elementor comes with a bunch of default pages layouts, Wordpress has many themes with default pages,...

> I saw that one was created with wix... maybe it's better and more professional than WP and Elementor

those web creator platforms use presets and they do a very good job. So if you can't build a site that looks professionally with WP, Elementor or any other page builder than I would use such a platform. I'm not a car mechanic so I don't start working on cars without learning it, I pay people to do it :wink: Perhaps that is something you should think about too. Those are business investments. E.g. all the research and work I'm doing here for free could be paid too. 

> Could you answer my first question about the look of the page?

No, can't do that. Don't know what you've build there :smile: Why do you want to use Aera theme? Switch to another one if that makes so many problems for you.

**Comment by dacunha31** [comment]
The header image. Your logo with the text above/below it and the text next to it. ==> it is a deliberate choice. The logo is colorful because it sticks to the very fashionable image of the spraying drone.
the whole composition. E.g. the white-blue gradients in the background of the headlines, the fade out colors in the red, yellow, green buttons on the right side ==> as I said before, the colors of the buttons will change. Regarding the gradients of the titles, they are voluntary because without any background, the text is not seen
Mentioned that before: look at tutorials. Elementor comes with a bunch of default pages layouts, Wordpress has many themes with default pages,... ==> Before the creation of this site, I looked at tutorials and what WP offered but I hadn't found anything that corresponded to what we wanted to do... It is true that I did not look at everything because there are many, so I let you give me a WP theme name that could fit
I'm not a car mechanic so I don't start working on cars without learning it, I pay people to do it  ==>If in life you only do the things you were trained to do... I pity you... it is important for me to discover worlds outside of our own because that is how we move forward in life. Moreover, I understand that your derogatory remarks are only made to make me understand that there is a profession of which you are a part and which charges to do this work. But if the costs could be included in the expenses of a company, the company must first be created... and in this case it is not the case, so we do as we can
No, can't do that. Don't know what you've build there 😄 Why do you want to use Aera theme? ==> I think more that you don't know how to do it, otherwise you would have answered and not ignored my request... Pride is a vice and sometimes it helps to question yourself rather than to denigrate.

The denigration that you show towards me is not part of your initial training as a webdesigner, I think. It is not because you think you master the subject on the fingertips, that you are allowed to address me with such contempt.
Criticism can sometimes be constructive, but in your case, it is made to hurt and try to make people who ask for help and who are not in the business understand not to acquire new experiences at the risk of "stealing" the work of designers.
I find this state of mind pathetic and I see that for this you respond to posts... yet nothing to do with Elementor, right? It just allows you to drown in your long answers the incompetence you show in consulting.

**Reply by m1ga** [reply]
Please read through the whole post again and see that I'm trying to help here and give you feedback what I would change and why. I'm not trying to be disrespectful or any thing.

>If in life you only do the things you were trained to do... I pity you.

I didn't say that, I said `without learning it`. So I would start learning to do that if I want to and not just start doing it without consulting others and listen to their help :wink: 

>  I think more that you don't know how to do it, otherwise you would have answered and not ignored my request

Again: You only show screenshots/videos of things you can do with Wordpress or Elementor in 100 different ways. How should I help you when I don't know how you've created those elements in the first place. That said: I found out how to fix your fade animation, I've helped you with the size. So please be a bit more respectful to me. 


You also have to see that you ask for basic web development questions here in a place that is dedicated to Elementor and if you read throught the discussions section here: you see many open questions with no reply. I've tried to be helpful here.

**Reply by dacunha31** [reply]
Please read through the whole post again and see that I'm trying to help here and give you feedback what I would change and why. I'm not trying to be disrespectful or any thing. ==> I take it that way because that's the feeling I have when I read the messages. If it is not the case, so much the better and I am delighted

I didn't say that, I said without learning it. So I would start learning to do that if I want to and not just start doing it without consulting others and listen to their help 😉 ==> that's what I'm trying to do... learn... but unconstructive criticism or things that don't make sense to a beginner don't advance my thinking. If you were more concrete in your messages, I would have better understood your intentions.

Again: You only show screenshots/videos of things you can do with Wordpress or Elementor in 100 different ways. ==> I don't know what else to show you... if you tell me, I'll make sure I do it, but if I'm not told, I'm not innate

That said: I found out how to fix your fade animation, I've helped you with the size ==> That's right, and I thank you for that. On the other hand, as it stands, it is still not satisfactory, even for a site from the 90s! As for the disrespect you feel, that is subjective, as is the disrespect I feel in your messages.

You also have to see that you ask for basic web development questions here in a place that is dedicated to Elementor ==> Yes because WP gave me the link to this site to find help... but it is not the objective

I think that for the purpose of this forum, it would have been useful to propose solutions to me to, for example, remove this logo in favor of something else or to rectify everything that does not fit, rather than giving me a lecture on the designer's job and the mandatory "payment" box. The forums are there to help, well, in my opinion

**Reply by m1ga** [reply]
Think we have to restart here as I didn't mean to be disrespectful in any way. 

I showed you pages which have a very decent design to look at. That's always a good start to create a own page. Get inspiration on what is out there. Told you already to change the fonts (more sans-serif, plain) or get rid of the gradient backgrounds. But that is just my choice, you don't have to do it. But if you want some feedback that is what I would give to you. You have to come up with solutions (e.g. don't use a background image in the content area or make the whole content area with a translucent white background). I wouldn't just put a white-blue gradient behind a text. That's what I meant with "from the 90s", because that's what we did there :smile: 
<img src="https://assets.justinmind.com/wp-content/uploads/2018/05/top-10-worst-90s-website-designs-3.png"/>

It is interesting that you've said `...because WP gave me the link to this site to find help`. Elementor has an own community at  https://forum.elementor.com/ with different categories for "gettings started", "design", "marketing",.... This github repo is mainly for the source code and discussions/bugs/issues around it. Not really the implementation. And sadly: not very much used at all (I wish they would close it and just use their own forum since that has more replies!)

> I don't know what else to show you... if you tell me, I'll make sure I do it, but if I'm not told, I'm not innate

Source code! Put the page somewhere where people can look at it and the code.

**Reply by dacunha31** [reply]
Hello,
Indeed, we will start again on good bases!
Thank you for your message.
How can I host my current site for free so that you can access it?
Thank you very much.

PS: I will not be able to make a page as you suggest!

**Reply by m1ga** [reply]
:thinking: don't know any free hosting, sorry. Check https://smartblogger.com/free-wordpress-hosting/ if there is something that helps (I just googled that page). It is not a recommendation since I don't know any of those services.

> PS: I will not be able to make a page as you suggest!

I'm sure you can once you get used to Wordpress/Elementor/Themes :+1: When you have the content in the page you can always do a facelift later on, so don't worry.

**Reply by dacunha31** [reply]
Hello,
I can't manage to publish the site so that you can have access to it and help me... I don't know how to go forward in the right way... except transferring my backups, I don't see how I can do otherwise....

**Comment by dacunha31** [comment]
Hello,
How are you?
I'm back after several months of taking your advice (and I thank you for it, because you were right) to redesign my site so that it looks better and more up-to-date! It is now published www.dso31.fr
Also, despite the fact that I put :
 .fadeInUp {
  animation-name: none !important;
}
as you advised me, the jumping problem continues...
You told me that with screenshots it was difficult for you to see why and to bring me a solution.
Now that the site is published, would you agree to help me solve this problem? I would be grateful.
Thank you very much.

**Reply by m1ga** [reply]
Hi! Doing good, hope you too.

We only removed the slide in animation with that line and that works fine. If you remove that you can see the slide in again from the bottom.

I guess you are talking about the resize in the content part now, right? All I can say there is that it is generated in the small layout and a JS is writing some inline styles on it to set a width and left coordinate:
![Screenshot_20220618_185843](https://user-images.githubusercontent.com/4334997/174449109-c318402f-402b-4e3e-9569-fd4b58325cb6.jpg)
(small overlay with the dark border shows the element after the page is fully loaded)

But I can't tell you where it comes from and why the original size is small. Maybe some bad coded mobile first in the theme? Perhaps the "fade-in" was hiding that before. 

Sorry that I can't give you any solution here. But you've managed to get the whole page online so I'm sure you'll find a solution to that too :+1: You could try to add a "fade in" animation in Elementor on the whole section. So that it will be hidden by default and fade in (not slide in as before) after half a second. That should be enough to cover up the JS

BTW: have a look at the GDPR guidelines in France. If it is the same as in Germany you have to block Google Fonts. Check https://de.wordpress.org/plugins/local-google-fonts/ or other "local font" plugins.

**Comment by aaliyahfraser** [comment]
Thanks for this article. You have shown some excellent plugins in this article. All the plugins are perfect. I have also posted a blog on this topic. This is because Elementor has so many free amazing add-ons. Easy-to-use features made creating websites a breeze. Here we talk about “Best Elementor Addons and Plugins for WordPress 2023 – Compared” You can check out it at – https://quomodosoft.com/best-elementor-addons-and-plugins/
